### PR TITLE
Fixing the typing package issue on py3.8 

### PR DESCRIPTION
**Pull request by awais786**
https://openedx.atlassian.net/browse/BOM-1587

Adding python3.8 support.
`typing package failing on py3.8`.

```
 File "/opt/python/3.8.0/lib/python3.8/runpy.py", line 183, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/opt/python/3.8.0/lib/python3.8/runpy.py", line 142, in _get_module_details
    return _get_module_details(pkg_main_name, error)
  File "/opt/python/3.8.0/lib/python3.8/runpy.py", line 109, in _get_module_details
    __import__(pkg_name)
  File "/home/travis/build/edx/XBlock/.tox/django22/lib/python3.8/site-packages/pytest/__init__.py", line 6, in <module>
    from _pytest.assertion import register_assert_rewrite
  File "/home/travis/build/edx/XBlock/.tox/django22/lib/python3.8/site-packages/_pytest/assertion/__init__.py", line 5, in <module>
    from typing import Any
  File "/home/travis/build/edx/XBlock/.tox/django22/lib/python3.8/site-packages/typing.py", line 1357, in <module>
    class Callable(extra=collections_abc.Callable, metaclass=CallableMeta):
  File "/home/travis/build/edx/XBlock/.tox/django22/lib/python3.8/site-packages/typing.py", line 1005, in __new__
    self._abc_registry = extra._abc_registry
AttributeError: type object 'Callable' has no attribute '_abc_registry' File "/opt/python/3.8.0/lib/python3.8/runpy.py", line 183, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/opt/python/3.8.0/lib/python3.8/runpy.py", line 142, in _get_module_details
    return _get_module_details(pkg_main_name, error)
  File "/opt/python/3.8.0/lib/python3.8/runpy.py", line 109, in _get_module_details
    __import__(pkg_name)
  File "/home/travis/build/edx/XBlock/.tox/django22/lib/python3.8/site-packages/pytest/__init__.py", line 6, in <module>
    from _pytest.assertion import register_assert_rewrite
  File "/home/travis/build/edx/XBlock/.tox/django22/lib/python3.8/site-packages/_pytest/assertion/__init__.py", line 5, in <module>
    from typing import Any
  File "/home/travis/build/edx/XBlock/.tox/django22/lib/python3.8/site-packages/typing.py", line 1357, in <module>
    class Callable(extra=collections_abc.Callable, metaclass=CallableMeta):
  File "/home/travis/build/edx/XBlock/.tox/django22/lib/python3.8/site-packages/typing.py", line 1005, in __new__
    self._abc_registry = extra._abc_registry
AttributeError: type object 'Callable' has no attribute '_abc_registry'
```

**Set 'typing' package to be installed only on Python 3.5. This fixes the import issue raised in Python 3.7 and 3.8 on using the 'typing' module.**

typing issue has some good discussion [here](https://github.com/python/typing/issues/573).

### CODE REVIEW

File: requirements/constraints.txt [hunk 1]
```diff
@@ -14,3 +14,6 @@
 # Adding this constraint till botocore can figure out a fix around this, keep a look at
 # https://github.com/boto/botocore/issues/1872
 python-dateutil<2.8.1
+
+
+typing; python_version == "3.5"
```
Comment (awais786) [inline requirements/constraints.txt:19]: typing is not working with python3.7 and python3.8

**Comment by nasthagiri** [comment]
Reminder to update the commit message and PR title to describe the change in this PR, beyond the JIRA ticket number. Thanks!

**Comment by awais786** [comment]
@morenol please review.

**Comment by morenol** [comment]
@awais786 We are ready for a new release, and entry in the changelog and do not allow failures on python 3.8

**Comment by morenol** [comment]
@awais786 also we need to add the classifiers for python 3.8

**Comment by awais786** [comment]
> @awais786 also we need to add the classifiers for python 3.8

already there.

**Comment by morenol** [comment]
My bad, now I think that this is ready to be merged

**Pull request merged**